crates/miden-objects/src/transaction/partial_blockchain.rs

```diff
@@ -182,6 +182,26 @@ impl PartialBlockchain {
         self.mmr.add(block_header.commitment(), track);
     }
 
+    /// Drop every block older than `cutoff_block_num` from the partial view.
+    ///
+    /// After the call all block headers with a number smaller than `cutoff_block_num` are gone and
+    /// their paths are no longer tracked in the inner partial MMR.
+    pub fn prune_before(&mut self, cutoff_block_num: BlockNumber) {
```

Comment:
Could consider using `RangeTo` aka `..cutoff_block_num` to make it self-descriptive.

```suggestion
    pub fn prune_to(&mut self, to: RangeTo<BlockNumber>) {
```

Reply:
One could also make it generic over ranges, which would allow combining both `prune` and `remove`. Unclear if that's useful though.